### ISSUE TITLE
post R 3.4.1 migration cleanup

### ISSUE THREAD

**Issue by daler** [bioconda/bioconda-recipes#6122]
As described in #6119, and mostly thanks to @bgruening, we are now building for R 3.4.1 and the majority of packages can be successfully built. However there still some remaining packages that are having trouble building. There does not seem to be a single solution to fix these, so it will take some time and effort to resolve problems on each package. 

The blacklist is here, and includes comments about the problems with each recipe:
https://github.com/bioconda/bioconda-recipes/blob/master/build-fail-blacklist. The goal is to have an empty blacklist.

This issue can serve as a central place to discuss and coordinate the work on fixing these packages.

**Comment by daler** [comment]
http://www.aroma-project.org/ says there was an update on 2017-09-13 addressing an issue with the `future` package. I'm starting to play around with updates to the various aroma packages and possibly the `future` package as well to see if that will resolve the issues.

**Comment by daler** [comment]
(xref #6126)

**Comment by daler** [comment]
For abyss, there's known incompatibility between older versions of abyss and later versions of boost: https://github.com/bcgsc/abyss/issues/130

**Comment by sebastian-luna-valero** [comment]
Hi @daler 

Congratulations for your great work!

Could you please confirm whether [peakranger](https://github.com/bioconda/bioconda-recipes/tree/master/recipes/peakranger) has been migrated to R 3.4.1?

It only has `r-base` as runtime dependency.

Best regards,
Sebastian

**Comment by daler** [comment]
It must be a boost or zlib version that's restricting it to R 3.3.2. I'll rebuild in a new PR.

**Comment by daler** [comment]
@sebastian-luna-valero see #6174

**Comment by sebastian-luna-valero** [comment]
Brilliant, thank you very much again!

**Comment by rvalieris** [comment]
hello @daler and @bgruening , I just noticed that a few packages are updated on the bioconda repo but not on the recipe, examples:

bioconductor-iranges is 2.8.2 on the recipe and 2.10.5 on the [repo](https://anaconda.org/bioconda/bioconductor-iranges/files)
bioconductor-biocparallel is 1.6.6 on the recipe and 1.10.1 on the [repo](https://anaconda.org/bioconda/bioconductor-biocparallel/files)
there is probably more like this, I haven't check too many yet.

anybody know whats going on here ? should I update these recipes anyway or leave it ?

**Comment by daler** [comment]
For the past couple of weeks I've been incrementally building packages on the `bulk` branch. The idea there is that the `bulk` branch builds recipes and uploads them immediately after a successful build. This allows a large number of recipes to be simultaneously built in the correct toposorted order until the timeout on travis . . . then the build is started again, but it starts where it left off.

So there are probably a _lot_ of such packages! The recipes will be updated once `bulk` is merged back into `master`.

I've been having a hard time getting `bioconductor-rhtslib` to compile, and there are other packages with various issues that have been holding me up. An example log is at https://travis-ci.org/bioconda/bioconda-recipes/jobs/288367238, if you want to help!

**Comment by rvalieris** [comment]
oh I see, its easy to miss the bulk branch !
I will look into it ! thanks.

**Comment by rvalieris** [comment]
it seems the uploads are failing on the bulk branch now:

    22:08:27 BIOCONDA ERROR UPLOAD ERROR: stdout+stderr: Using Anaconda API:
    binstar_client.errors.Unauthorized: ('Authentication token has expired', 401)

**Comment by daler** [comment]
@johanneskoester @bgruening I seem to remember having to choose a very particular set of options when generating the anaconda API token to avoid permission errors. Do either of you have notes on that?

**Comment by johanneskoester** [comment]
@daler sorry, no notes. "Allow all operations"? Or everything except the pypi stuff?

**Comment by daler** [comment]
I seem to remember the right settings were non-intuitive. Trial and error, I guess. I'll play with it on the bulk branch.

**Comment by daler** [comment]
Ah wait, we're not using encrypted vars in .travis.yml, rather we're using vars added to travis-ci interface. That should be easier to play with, but more disruptive to running jobs.

However, when tracking down what time the token expired, I now see that this (circexplorer) ran 12 minutes ago, and successfully uploaded a package: https://travis-ci.org/bioconda/bioconda-recipes/jobs/289507799.

So maybe it's something going on in the bulk branch? I will look into this later today.

**Comment by bgruening** [comment]
@daler @johanneskoester mh ... sorry to have not informed you, I saw an error on master this morning and created already a new API key. I did rerun the failing jobs on master and packages were indeed pushed ... This is wired. Is it still broken. I allowed `all operations`

**Comment by rvalieris** [comment]
in that case I will just restart the jobs then, the logs I saw were from last night.

**Comment by bgruening** [comment]
@rvalieris yes please restart. I hope this was fixed around 11h ago.

**Comment by daler** [comment]
Thanks Bjoern!

**Comment by bgruening** [comment]
I think this can be also closed. Next migration comes soon ;)
